server_action_mass_edit/wizard/mass_editing_wizard.py

```diff
@@ -0,0 +1,335 @@
+            "type": "selection",
+            "string": field_info["string"],
+            "selection": selection,
+        }
+        # Add field info
+        result[field.name] = field_info
+        return result
+
+    @api.model
+    def _insert_field_in_arch(self, line, field, main_xml_group):
+        etree.SubElement(
+            main_xml_group,
+            "label",
+            {
+                "for": "selection__" + field.name,
+            },
+        )
+        div = etree.SubElement(
+            main_xml_group,
+            "div",
+            {
+                "class": "d-flex",
+            },
+        )
+        etree.SubElement(
+            div,
+            "field",
+            {
+                "name": "selection__" + field.name,
+                "modifiers": '{"required": true}',
+                "class": "w-25",
+            },
+        )
+        field_vals = self._get_field_options(field)
+        if line.widget_option:
+            field_vals["widget"] = line.widget_option
+        field_element = etree.SubElement(div, "field", field_vals)
+        if field.ttype == "one2many":
+            comodel = self.env[field.relation]
+            dummy, form_view = comodel._get_view(view_type="form")
+            dummy, tree_view = comodel._get_view(view_type="tree")
+            field_context = {}
+            if form_view:
+                field_context["form_view_ref"] = form_view.id
+            if tree_view:
+                field_context["tree_view_ref"] = tree_view.id
+            if field_context:
+                field_element.attrib["context"] = json.dumps(field_context)
+            else:
+                model_arch, dummy = self.env[field.model]._get_view(view_type="form")
+                embedded_tree = None
+                for node in model_arch.xpath(f"//field[@name='{field.name}'][./tree]"):
+                    embedded_tree = node.xpath("./tree")[0]
+                    break
+                if embedded_tree is not None:
+                    for node in embedded_tree.xpath("./*"):
+                        modifiers = node.get("modifiers")
+                        if modifiers:
+                            node.attrib["modifiers"] = modifiers
+                    field_element.insert(0, embedded_tree)
+
+        return field_element
+
+    def _get_field_options(self, field):
+        return {
+            "name": field.name,
+            "modifiers": '{"invisible": "\
+            "[["selection__%s", "in", ["ignore", "remove"]]]}'
+            % field.name,
+            "class": "w-75",
+        }
+
+    @api.model
+    def get_views(self, views, options=None):
+        for view, type in views:
+            if view:
+                view = self.env["ir.ui.view"].browse(view)
+                server_action = view.mass_server_action_id
+                self = self.with_context(server_action_id=server_action.id)
+        return super().get_views(views, options)
```

Comment:
I think replace type by _type should solve pre-commit warning

Reply:
@CRogos Thaks! Now all test had pass.
Best regards